cranelift-codegen/src/isa/x86/fde.rs

```diff
@@ -252,7 +251,13 @@ pub fn emit_fde(func: &Function, isa: &dyn TargetIsa, sink: &mut dyn FrameUnwind
     for (off, r) in relocs {
         sink.reloc(r, off + unwind_start);
     }
-    let fde_offset = unsafe { ptr::read::<u32>(bytes.as_ptr() as *const u32) } as usize + 4;
+
+    let fde_first_u32 = u32::from_le_bytes(
+        bytes.as_slice()[..4]
+            .try_into()
+            .expect("4 bytes at the beginning of the allocated FDE"),
+    );
+    let fde_offset = fde_first_u32 as usize + 4;
```

Comment:
@iximeow, @yurydelendik: here it is! (With an extra variable assignment)

Reply:
@yurydelendik Do you recall what this u32 specifically is? It's a count of FDEs or something, right? It'd be nice if we could use that name instead of `fde_first_u32`.

Reply:
As discussed [in previous PR](https://github.com/bytecodealliance/cranelift/pull/1320#discussion_r366607644), gimli does not provide right API to get all data needed. It is a field (u32) that show length of the CIE, so the len and 4 u32 bytes is offset of the FDE structure that follows CIE.

Reply:
Should I rename `fde_first_u32` as `cie_len`?

Reply:
Nit: the expect message is wrong. That `try_into` isn't checking there were 4 bytes at the beginning of `bytes`. It's checking that the slice passed to `try_into` is 4 bytes, which will always be true because we indexed with a range of `[..4]`. If there were less than 4 bytes, then there would be a panic on the slice indexing, not on the `try_into().expect()`. IMO using `unwrap` is fine in this case.